src/components/FileMachineProvider.tsx

```diff
@@ -271,6 +274,38 @@ export const FileMachineProvider = ({
     }
   )
 
+  const kclCommandMemo = useMemo(
+    () =>
+      kclCommands(async (data) => {
+        if (data.method === 'overwrite') {
+          codeManager.updateCodeStateEditor(data.code)
+          await kclManager.executeCode(true)
+          await codeManager.writeToFile()
+        } else if (data.method === 'newFile' && isDesktop()) {
+          send({
+            type: 'Create file',
+            data: {
+              name: data.sampleName,
+              content: data.code,
+              makeDir: false,
+            },
+          })
+        }
+      }),
```

Comment:
In future we should maybe add this as a flow in the `fileMachine`, and then do a `fileMachineCommandConfig` and the whole machine-powered commands for the file tree, but this was a lightweight step toward that, passing in the stuff that needed to have React things available as an `onSubmit` callback.